File: solutions/2582. Pass the Pillow/2582.cpp
// Time complexity: O(1)
// Space complexity: O(1)
class Solution {
 public:
  int passThePillow(int n, int time) {
    time %= (n - 1) * 2;
    if (time < n)
      return 1 + time;
    return n - (time - (n - 1));
  }
};

File: solutions/2582. Pass the Pillow/2582.py
# Time complexity: O(1)
# Space complexity: O(1)
class Solution:
  def passThePillow(self, n: int, time: int) -> int:
    time %= (n - 1) * 2
    if time < n:
      return 1 + time
    return n - (time - (n - 1))